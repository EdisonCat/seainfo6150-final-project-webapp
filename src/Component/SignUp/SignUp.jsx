import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import styles from "./SignUp.module.css";
import MarvelFandomContext from "../../Context/MarvelFandomContext";
export default function SignUp() {
	const [email, setEmail] = useState("");
	const [username, setUsername] = useState("");
	const [password, setPassword] = useState("");
	const [confirm, setConfirm] = useState("");
	const [type, setType] = useState("Normal");
	const [agreement, setAgreement] = useState(false);
	const [event, setEvent] = useState(true);
	const [tv, setTv] = useState(true);
	const [comics, setComics] = useState(true);
	const [movies, setMovies] = useState(true);
	const [join, setJoin] = useState("Marvel and Medicine");
	const history = useHistory();
	const context = useContext(MarvelFandomContext);
	const emailPattern = new RegExp(
		/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/
	);
	const handleClick = async (e) => {
		e.preventDefault();
		if (!emailPattern.test(email)) {
			alert("Please enter a valid email address!");
			return;
		} else if (username.length <= 3) {
			alert("Username must be longer than 3 characters!");
			return;
		} else if (password.length < 6) {
			alert("Password must be longer than 6 digits!");
			return;
		} else if (password !== confirm) {
			alert("Please check your password!");
			return;
		} else if (!type) {
			alert("Please the type of account you'd like to sign up for!");
			return;
		} else if (!agreement) {
			alert("Please read the agreement!");
			return;
		}
		const res = await fetch("https://demo9528898.mockable.io/signup", {
			method: "POST",
			headers: { "Content-Type": "application/json" },
			body: {
				email: email,
				username: username,
				password: password,
			},
		});
		if (res.status === 200) {
			context.login(username);
			localStorage.setItem("mf_username", username);
			localStorage.setItem("mf_email", email);
			localStorage.setItem("mf_pass", password);
			history.push({
				pathname: "/signup/success",
				state: {
					username: username,
					type: type,
					event: event,
					comics: comics,
					tv: tv,
					movies: movies,
					join: join,
				},
			});
		}
	};
	return (
		<div className={styles.container}>
			<form>
				<input
					className={styles.inputtext}
					type="text"
					placeholder="E-mail"
					onChange={(e) => setEmail(e.target.value)}
				/>
				<input
					className={styles.inputtext}
					type="text"
					placeholder="Username"
					onChange={(e) => setUsername(e.target.value)}
				/>
				<input
					className={styles.inputtext}
					type="password"
					placeholder="Password"
					onChange={(e) => setPassword(e.target.value)}
				/>
				<input
					className={styles.inputtext}
					type="password"
					placeholder="Confirm your password"
					onChange={(e) => setConfirm(e.target.value)}
				/>
				<label htmlFor="type">
					Type of Account:
					<select id="type" onChange={(e) => setType(e.target.value)}>
						<option value="normal">Normal</option>
						<option value="premium">Premium</option>
						<option value="gold">Gold</option>
						<option value="Diamond">Diamond</option>
					</select>
				</label>
				<label htmlFor="join">
					Join an ongoing event:
					<select id="join" onChange={(e) => setJoin(e.target.value)}>
						<option value="">Not interested</option>
						<option value="Marvel and Medicine">
							Marvel and Medicine
						</option>
						<option value="Marvel for Homeless">
							Marvel for Homeless
						</option>
						<option value="Marvel Foundation">
							Marvel Foundation
						</option>
						<option value="Marvel TED Talk">Marvel TED Talk</option>
					</select>
				</label>

				<fieldset>
					I want to subscribe to notification emails for
					<label htmlFor="events">
						<input
							className={styles.check}
							id="events"
							defaultChecked={true}
							type="checkbox"
							onChange={() => setEvent(!event)}
						/>
						Events
					</label>
					<label htmlFor="comics">
						<input
							className={styles.check}
							id="comics"
							defaultChecked={true}
							type="checkbox"
							onChange={() => setComics(!comics)}
						/>
						Comics
					</label>
					<label htmlFor="tv">
						<input
							className={styles.check}
							id="tv"
							defaultChecked={true}
							type="checkbox"
							onChange={() => setTv(!tv)}
						/>
						TV Shows
					</label>
					<label htmlFor="movies">
						<input
							className={styles.check}
							id="movies"
							defaultChecked={true}
							type="checkbox"
							onChange={() => setMovies(!movies)}
						/>
						Movies
					</label>
				</fieldset>
				<label htmlFor="check">
					I agree to the&nbsp;
					<a
						href="https://disneytermsofuse.com/english/"
						target="_blank"
						rel="noopener noreferrer"
					>
						Terms of Service
					</a>
					&nbsp;and&nbsp;
					<a
						href="https://privacy.thewaltdisneycompany.com/en/current-privacy-policy/"
						target="_blank"
						rel="noopener noreferrer"
					>
						Privacy Policy
					</a>
					.
					<input
						className={styles.check}
						defaultChecked={false}
						id="check"
						type="checkbox"
						onChange={() => setAgreement(!agreement)}
					/>
				</label>
			</form>
			<button onClick={(e) => handleClick(e)}>Sign Up</button>
		</div>
	);
}