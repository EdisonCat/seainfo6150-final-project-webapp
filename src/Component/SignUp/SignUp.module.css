@import url("https://fonts.googleapis.com/css2?family=Work+Sans:wght@300&display=swap");
.container {
	display: flex;
	flex-direction: column;
	min-height: 900px;
	background-color: #333;
	border-top: 1px solid grey;
	justify-content: center;
	align-items: center;
	color: white;
	font-family: "Work Sans", sans-serif;
}
.container form {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}
.inputtext {
	margin: 5px;
	border-radius: 30px;
	width: 200px;
	height: 40px;
	outline: none;
	padding: 5px;
	font-size: 15px;
	transition: 0.5s;
	border: none;
}
.container select {
	outline: none;
	width: 100px;
	border-radius: 50px;
	margin: 5px;
}
.container label {
	display: flex;
	align-items: center;
}
.container button {
	border: none;
	margin: 5px;
	width: 80px;
	height: 30px;
	border-radius: 30px;
	transition: 0.5s;
	cursor: pointer;
	outline: none;
	min-width: fit-content;
	background: white;
}
.inputtext:hover,
.inputtext:focus {
	width: 220px;
}
.container button:hover {
	width: 100px;
	color: white;
	background: red;
	border: none;
}
a {
	color: white;
	outline: none;
	text-decoration: underline;
	text-decoration-color: #333;
	transition: 0.5s;
}
a:hover {
	text-decoration-color: red;
}
fieldset {
	border: none;
	display: flex;
	flex-direction: column;
}
i {
	text-decoration: underline;
	text-decoration-color: #333;
	transition: 0.5s;
	cursor: pointer;
}
i:hover {
	text-decoration-color: red;
}