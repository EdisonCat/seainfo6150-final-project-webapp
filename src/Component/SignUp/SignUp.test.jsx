import React from "react";
import SignUp from "./SignUp";
it("rendered SignUp correctly", () => {
	// eslint-disable-next-line no-undef
    const { container } = render(<SignUp />);
    expect(container).toMatchSnapshot();
});