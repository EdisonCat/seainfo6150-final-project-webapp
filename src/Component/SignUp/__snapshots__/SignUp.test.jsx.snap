// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered SignUp correctly 1`] = `
<div>
  <div
    class="container"
  >
    <form>
      <input
        class="inputtext"
        placeholder="E-mail"
        type="text"
      />
      <input
        class="inputtext"
        placeholder="Username"
        type="text"
      />
      <input
        class="inputtext"
        placeholder="Password"
        type="password"
      />
      <input
        class="inputtext"
        placeholder="Confirm your password"
        type="password"
      />
      <label
        for="type"
      >
        Type of Account:
        <select
          id="type"
        >
          <option
            value="normal"
          >
            Normal
          </option>
          <option
            value="premium"
          >
            Premium
          </option>
          <option
            value="gold"
          >
            Gold
          </option>
          <option
            value="Diamond"
          >
            Diamond
          </option>
        </select>
      </label>
      <label
        for="join"
      >
        Join an ongoing event:
        <select
          id="join"
        >
          <option
            value=""
          >
            Not interested
          </option>
          <option
            value="Marvel and Medicine"
          >
            Marvel and Medicine
          </option>
          <option
            value="Marvel for Homeless"
          >
            Marvel for Homeless
          </option>
          <option
            value="Marvel Foundation"
          >
            Marvel Foundation
          </option>
          <option
            value="Marvel TED Talk"
          >
            Marvel TED Talk
          </option>
        </select>
      </label>
      <fieldset>
        I want to subscribe to notification emails for
        <label
          for="events"
        >
          <input
            checked=""
            class="check"
            id="events"
            type="checkbox"
          />
          Events
        </label>
        <label
          for="comics"
        >
          <input
            checked=""
            class="check"
            id="comics"
            type="checkbox"
          />
          Comics
        </label>
        <label
          for="tv"
        >
          <input
            checked=""
            class="check"
            id="tv"
            type="checkbox"
          />
          TV Shows
        </label>
        <label
          for="movies"
        >
          <input
            checked=""
            class="check"
            id="movies"
            type="checkbox"
          />
          Movies
        </label>
      </fieldset>
      <label
        for="check"
      >
        I agree to the 
        <a
          href="https://disneytermsofuse.com/english/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Terms of Service
        </a>
         and 
        <a
          href="https://privacy.thewaltdisneycompany.com/en/current-privacy-policy/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Privacy Policy
        </a>
        .
        <input
          class="check"
          id="check"
          type="checkbox"
        />
      </label>
    </form>
    <button>
      Sign Up
    </button>
  </div>
</div>
`;
